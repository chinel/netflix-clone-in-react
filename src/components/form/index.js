import React from "react";
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from "./styles/form";

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};
