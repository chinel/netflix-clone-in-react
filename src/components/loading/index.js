import React from "react";

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner>
      <LockBody />
      <Picture src={`/images/users/${src}.png`}
    </Spinner>
  );
}
