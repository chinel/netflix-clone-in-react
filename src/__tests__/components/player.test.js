import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Player } from "../../components";

describe("<Player />", () => {
  it("render the <Player /> with a bunny video", () => {
    const { container, getByText, queryByTestId } = render(
      <Player>
        <Player.Button />
        <Player.Video src="/videos/bunny.mp4" />
      </Player>
    );

    expect(queryByTestId("player")).toBeFalsy();
    fireEvent.click(getByText("Play"));

    expect(queryByTestId("player")).toBeTruthy();
    fireEvent.click(queryByTestId("player")); //this clicks the player component which is more of like click outside the component and it sets it to falsy

    expect(queryByTestId("player")).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
