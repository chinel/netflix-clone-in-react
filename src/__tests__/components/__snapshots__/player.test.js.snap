// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> render the <Player /> with a bunny video 1`] = `
<div
  class="player__Container-sc-4yj2e3-0"
>
  <button
    class="player__Button-sc-4yj2e3-4 jlZLqX"
  >
    Play
  </button>
</div>
`;
